drivers/console/ram_console.c

```diff
@@ -11,23 +12,36 @@
 #include <zephyr/sys/printk.h>
 #include <zephyr/device.h>
 #include <zephyr/init.h>
+#include <zephyr/linker/devicetree_regions.h>
 
 extern void __printk_hook_install(int (*fn)(int));
 extern void __stdout_hook_install(int (*fn)(int));
 
+#if DT_HAS_CHOSEN(zephyr_ram_console)
+char ram_console[DT_REG_SIZE(DT_CHOSEN(zephyr_ram_console))]
+__attribute__((__section__(LINKER_DT_NODE_REGION_NAME(DT_CHOSEN(zephyr_ram_console)))));
+#else
 /* Extra byte to ensure we're always NULL-terminated */
 char ram_console[CONFIG_RAM_CONSOLE_BUFFER_SIZE + 1];
+#endif
+
 static int pos;
+static unsigned int size;
 
 static int ram_console_out(int character)
 {
 	ram_console[pos] = (char)character;
-	pos = (pos + 1) % CONFIG_RAM_CONSOLE_BUFFER_SIZE;
+	pos = (pos + 1) % size;
 	return character;
 }
 
 static int ram_console_init(void)
 {
+#if DT_HAS_CHOSEN(zephyr_ram_console)
+	size = DT_REG_SIZE(DT_CHOSEN(zephyr_ram_console)) - 1;
+#else
+	size = CONFIG_RAM_CONSOLE_BUFFER_SIZE;
```

Comment:
Not a big deal, but note that both of these values which can be assigned to size are compile-time constants, so you're adding needless bytes to .data and an extra load per character output when you could have just done this using the preprocessor.

Reply:
Thanks for your comments!
Updated to use preprocessor.